src/app/map.js

```diff
@@ -5,95 +5,46 @@ define([
-				      content: '<span class="infoTemplateContentRowLabel">Date: </span>' +
-				          '<span class="infoTemplateContentRowItem">${requestdate:DateFormat}</span><br><span class="infoTemplateContentRowLabel">Phone: </span>' +
-				          '<span class="infoTemplateContentRowItem">${phone:formatPhoneNumber}</span><br><span class="infoTemplateContentRowLabel">Name: </span>' +
-				          '<span class="infoTemplateContentRowItem">${name}</span><br><span class="infoTemplateContentRowLabel">Severity: </span>' +
-				          '<span class="infoTemplateContentRowItem">${severity:severityDomainLookup}</span><br><span class="infoTemplateContentRowLabel">Type: </span>' +
-				          '<span class="infoTemplateContentRowItem">${requesttype:requestTypeDomainLookup}</span><br><span class="infoTemplateContentRowLabel">Comments: </span>' +
-				          '<span class="infoTemplateContentRowItem">${comment}</span>'
-				      },
-		  		severityFieldDomainCodedValuesDict: 
-		  			 {   
-		  				'0': 'General Nuisance',
-    				    '1': 'Important To Resolve Soon',
-    					'2': 'Critical Issue'
-		  			 },
-		  		requestTypeFieldDomainCodedValuesDict:
-		  			 {
-		  			 	'0': 'Abandoned Vehicle',
-					    '1': 'Animal Services',
-					    '2': 'Driveway Infraction',
-					    '3': 'Flooding',
-					    '4': 'Graffiti Removal',
-					    '5': 'Homeless Nuisance',
-					    '6': 'Illegal Dumping',
-					    '7': 'Parking Violation',
-					    '8': 'Plant/Tree Complaint',
-					    '9': 'Pothole Obstruction',
-					    '10': 'Roadway Danger',
-					    '11': 'Sidewalk Danger',
-					    '12': 'Streetlight Broken',
-					    '13': 'Street Sign Missing/Damaged',
-					    '14': 'Trash Removal',
-					    '15': 'Water Leak',
-					    '16': 'Yard Waste Removal'
-		  			},
-
 		  		addFeatureLayer: function(){
-		  			var citizenRequestLayer = new FeatureLayer(this.citizenRequestLayerUrl, {
+		  			var citizenRequestLayer = new FeatureLayer(config.citizenRequestLayerUrl, {
                     mode: FeatureLayer.MODE_ONEDEMAND,
-                    infoTemplate: new InfoTemplate(this.infoTemplate),
+                    infoTemplate: new InfoTemplate(config.infoTemplate),
                     outFields: ['*']
                 });
 		  			return citizenRequestLayer;
 		  		},	
 		  		addLegend: function(map, location){
-		  				  location = typeof location !== 'undefined' ? location : 'legend';
-			  			  var legend = new Legend({
-						        map: map
-						    }, location);
-			  			legend.startup();  	
-			  			return legend;
+		  			config.legendOptions.map = typeof map !== 'undefined' ? map : null;
+		  			location = typeof location !== 'undefined' ? location : 'legend';
+			  		var legend = new Legend(config.legendOptions, location);
+			  		legend.startup();  	
+			  		return legend;
 		  		},
 		  		addGeocoder:function(map, location){
+		  			config.geocoderOptions.map = typeof map !== 'undefined' ? map : null;
```

Comment:
Don't set the `map` property of the config object. Instead create a local object that mixes in options from the config file:

```js
var opt = lang.mixin(config.geocoderOptions, {map: map});
```

See: http://dojotoolkit.org/reference-guide/1.10/dojo/_base/lang.html#mixin

Please do the same w/ `config.legendOptions` and `config.locateButtonOptions`.